internal/manifest/oci_index.go

```diff
@@ -33,6 +34,74 @@ type OCI1IndexPublic struct {
 	imgspecv1.Index
 }
 
+type manifestLessFunc func(m1, p2 *imgspecv1.Descriptor) bool
+
+type manifestsSorter struct {
+	manifests []imgspecv1.Descriptor
+	less      []manifestLessFunc
+}
+
+// Extend following function to support other annotations
+func manifestByAnnotation(m1, m2 *imgspecv1.Descriptor) bool {
+	m1IsZstd := false
+	m2IsZstd := false
+	isZstd, ok := m1.Annotations["io.github.containers.compression.zstd"]
+	if ok {
+		if isZstd == "true" {
+			m1IsZstd = true
+		}
+	}
+	isZstd, ok = m2.Annotations["io.github.containers.compression.zstd"]
+	if ok {
+		if isZstd == "true" {
+			m2IsZstd = true
+		}
+	}
+	return !m2IsZstd && m1IsZstd
```

Comment:
Doesn’t this sort Zstd before gzip? (Ultimately there should be a test of `EditInstances`)

For compatibility, IIRC we want gzip first, so that old implementations use that one.